backend/corpora/common/entities/dataset.py

```diff
@@ -155,13 +155,9 @@ def create_revision(self) -> "Dataset":
         return Dataset(revision_dataset)
 
     def tombstone_dataset_and_delete_child_objects(self):
-        self.update(tombstone=True)
+        self.update(tombstone=True, deployment_directories=[], artifacts=[])
```

Comment:
It can lead to dangling artifacts, but only if all datasets that reference these artifacts become a tombstone or are deleted without cleanup up the S3 artifacts.

Reply:
The goal of this function is to create dangling artifacts as far as the collection revision is concerned. The public published collection should still reference the artifacts.

Reply:
would it worth making the difference in handling explicit? `tombstone_dataset_and_delete_child_objects_public_collection` and `tombstone_dataset_and_delete_child_objects_private_collection`

Reply:
the `tombstone_dataset_and_delete_child_objects_public_collection` would be need until we do publish a tombstone collection so maybe in the near future.